enclave-runtime/src/attestation.rs

```diff
@@ -77,9 +77,7 @@ pub const REPORT_SUFFIX: &str = "/sgx/dev/attestation/v4/report";
 pub unsafe extern "C" fn get_mrenclave(mrenclave: *mut u8, mrenclave_size: u32) -> sgx_status_t {
 	let mrenclave_slice = slice::from_raw_parts_mut(mrenclave, mrenclave_size as usize);
 
-	let ocall_api = OCallComponentFactory::attestation_api();
-
-	match ocall_api.get_mrenclave_of_self() {
+	match OcallApi.get_mrenclave_of_self() {
```

Comment:
The reason why I did it the way it was, is to allow better abstraction for the future, in case we find a better way to inject those objects into the function. 
It's probably fine for now, but we've basically 'reverted' to a less generic version that will not allow test mocking.

Reply:
This is actually something I still don't get. I don't see any way how to inject things here. We had the concrete factory before there before. How would you make it return something different?

Reply:
Something you see in dependency injection for instance, is that you have a node/container that can serve any trait instance you'd like. That container in this case would have to be some global instance. In tests, you can initialize the global container with mock instances, so that in this function here, we'd get those in a test setup.

Reply:
Ok, I think I understand now. Thanks for the explanation!